arwen/elrondapitest/bigFloats_test.go

```diff
@@ -0,0 +1,690 @@
+			WithArguments(repsArgument,
+				floatArgument1).
+			Build()).
+		AndAssertResults(func(host arwen.VMHost, stubBlockchainHook *contextmock.BlockchainHookStub, verify *test.VMOutputVerifier) {
+			bigFloatValue := new(big.Float)
+			_ = bigFloatValue.GobDecode(floatArgument1)
+			for i := 0; i < numberOfReps; i++ {
+				bigFloatValue.Add(bigFloatValue, bigFloatValue)
+			}
+			floatBuffer, _ := bigFloatValue.GobEncode()
+			verify.
+				Ok().
+				ReturnData(
+					floatBuffer)
+		})
+}
+
+func TestBigFloats_Sub(t *testing.T) {
+	test.BuildInstanceCallTest(t).
+		WithContracts(
+			test.CreateInstanceContract(test.ParentAddress).
+				WithCode(test.GetTestSCCode("big-floats", "../../"))).
+		WithInput(test.CreateTestContractCallInputBuilder().
+			WithGasProvided(100000).
+			WithFunction("BigFloatSubTest").
+			WithArguments([]byte{0, 0, 0, byte(10)},
+				floatArgument1).
+			Build()).
+		AndAssertResults(func(host arwen.VMHost, stubBlockchainHook *contextmock.BlockchainHookStub, verify *test.VMOutputVerifier) {
+			bigFloatValue := big.NewFloat(0)
+			_ = bigFloatValue.GobDecode(floatArgument1)
+			for i := 0; i < 10; i++ {
+				bigFloatValue.Sub(bigFloatValue, bigFloatValue)
+			}
+			floatBuffer, _ := bigFloatValue.GobEncode()
+			verify.
+				Ok().
+				ReturnData(
+					floatBuffer)
+		})
+}
+
+func TestBigFloats_Success_Mul(t *testing.T) {
+	numberOfReps := 10
+	repsArgument := []byte{0, 0, 0, byte(numberOfReps)}
+	test.BuildInstanceCallTest(t).
+		WithContracts(
+			test.CreateInstanceContract(test.ParentAddress).
+				WithCode(test.GetTestSCCode("big-floats", "../../"))).
+		WithInput(test.CreateTestContractCallInputBuilder().
+			WithGasProvided(100000).
+			WithFunction("BigFloatMulTest").
+			WithArguments(repsArgument,
+				floatArgument1).
+			Build()).
+		AndAssertResults(func(host arwen.VMHost, stubBlockchainHook *contextmock.BlockchainHookStub, verify *test.VMOutputVerifier) {
+			bigFloatValue := new(big.Float)
+			err := bigFloatValue.GobDecode(floatArgument1)
+			require.Nil(t, err)
+			for i := 0; i < numberOfReps; i++ {
+				resultMul := new(big.Float).Mul(bigFloatValue, bigFloatValue)
+				bigFloatValue.Set(resultMul)
+			}
+			floatBuffer, _ := bigFloatValue.GobEncode()
+			fmt.Println(floatBuffer)
+			verify.
+				Ok().
+				ReturnData(
+					floatBuffer)
+		})
+}
+
+func TestBigFloats_FailExecution_Mul(t *testing.T) {
+	numberOfReps := 30
+	repsArgument := []byte{0, 0, 0, byte(numberOfReps)}
+	test.BuildInstanceCallTest(t).
+		WithContracts(
+			test.CreateInstanceContract(test.ParentAddress).
+				WithCode(test.GetTestSCCode("big-floats", "../../"))).
+		WithInput(test.CreateTestContractCallInputBuilder().
+			WithGasProvided(100000).
+			WithFunction("BigFloatMulTest").
+			WithArguments(repsArgument,
+				floatArgument1).
+			Build()).
+		AndAssertResults(func(host arwen.VMHost, stubBlockchainHook *contextmock.BlockchainHookStub, verify *test.VMOutputVerifier) {
+			verify.
+				ReturnCode(10).
+				ReturnData()
```

Comment:
Checking `ReturnData` when execution has failed is not needed - it's always empty.
```suggestion
			verify.ExecutionFailed()
```